### PR TITLE
bundle of changes to up the overall functionality of site

### PR DESCRIPTION
**Pull request by roccopolimen**
Lot's of changes needed to increase the overall usability of the site. Brief summary
* calendar works now when cards don't have dueDates
* fix parenthese problem in app.js so profile page can be accessed
* all modals on board-page open properly (did not implement the closing, as not sure how that works with the css thing).
* made the client side code for board-page.js more modular
* comment modal form submission is finished
* fixed the routes that load partials, to set layout: null
* slight changes to many handlebars to make sure they are structured how we want them to be

### CODE REVIEW

**Comment by jwolfcrafter** [comment]
LGTM

File: src/views/board.handlebars [hunk 1]
```diff
@@ -66,7 +66,7 @@
         {{/each}}
         <form id="new-list" method="POST" action="/board/{{board._id}}">
             <label for="newListNameInput" hidden>List: </label>
-            <input type="text" name="listName" id="newListNameInput">
+            <input type="text" name="listName" id="newListNameInput" required/>
```
Comment (bwormser) [inline src/views/board.handlebars:69]: I don' think the slash is necessary after required, but if it works it doesn't matter

File: src/views/partials/comment.handlebars [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    <div class="comment-container">
+        <h3>{{user.initials}}</h3>
+        <p class="comment">{{comment}}</p>
+    </div>
+{{/each}}
+<form id="comment-form" action="/board/card/comments/{{boardId}}/{{cardId}}" method="POST">
+    <input type="hidden" name="_method" value="PUT" />
+     <label>
+        Comment
+        <input type="text" name="comment" class="comment-box" id="commentInput" required/>
```
Comment (bwormser) [inline src/views/partials/comment.handlebars:12]: Same thing as before, don't think required/ is necessary

Comment (bwormser) [reply src/views/partials/comment.handlebars:12]: *Don't think the / is necessary

**Pull request merged**